File: src/main/java/priv/bajdcc/LALR1/syntax/lexer/tokenizer/PropertyTokenizer.java
package priv.bajdcc.LALR1.syntax.lexer.tokenizer;

import priv.bajdcc.util.lexer.algorithm.TokenAlgorithm;
import priv.bajdcc.util.lexer.algorithm.filter.StringPairFilter;
import priv.bajdcc.util.lexer.error.RegexException;
import priv.bajdcc.util.lexer.regex.IRegexStringIterator;
import priv.bajdcc.util.lexer.token.MetaType;
import priv.bajdcc.util.lexer.token.Token;
import priv.bajdcc.util.lexer.token.TokenType;

/**
 * 终结符解析
 *
 * @author bajdcc
 */
public class PropertyTokenizer extends TokenAlgorithm {

	public PropertyTokenizer() throws RegexException {
		super(getRegexString(), new StringPairFilter(MetaType.LBRACE, MetaType.RBRACE));
	}

	public static String getRegexString() {
		return "{.*}";
	}

	/* （非 Javadoc）
	 * @see priv.bajdcc.lexer.algorithm.ITokenAlgorithm#getToken(java.lang.String, priv.bajdcc.lexer.token.Token)
	 */
	@Override
	public Token getToken(String string, Token token, IRegexStringIterator iterator) {
		token.kToken = TokenType.MACRO;
		token.object = string;
		return token;
	}
}

File: src/main/java/com/bajdcc/LALR1/syntax/lexer/tokenizer/PropertyTokenizer.kt
package com.bajdcc.LALR1.syntax.lexer.tokenizer

import com.bajdcc.util.lexer.algorithm.TokenAlgorithm
import com.bajdcc.util.lexer.algorithm.filter.StringPairFilter
import com.bajdcc.util.lexer.error.RegexException
import com.bajdcc.util.lexer.regex.IRegexStringIterator
import com.bajdcc.util.lexer.token.MetaType
import com.bajdcc.util.lexer.token.Token
import com.bajdcc.util.lexer.token.TokenType

/**
 * 终结符解析
 *
 * @author bajdcc
 */
class PropertyTokenizer @Throws(RegexException::class)
constructor() : TokenAlgorithm(regexString, StringPairFilter(MetaType.LBRACE, MetaType.RBRACE)) {

    override fun getToken(string: String, token: Token, iterator: IRegexStringIterator): Token? {
        token.type = TokenType.MACRO
        token.obj = string
        return token
    }

    companion object {

        val regexString: String
            get() = "{.*}"
    }
}
